app/grandchallenge/components/form_fields.py

```diff
@@ -43,75 +48,130 @@ def __init__(
-                kwargs["initial"] = initial.value
-        elif initial is not None:
-            # in the AlgorithmJobCreateForm,
-            # the initial value is the default_value for the interface,
-            # not an existing CIV
-            kwargs["initial"] = initial
+                return self.initial.value
+        elif (
+            isinstance(self.initial, ComponentInterfaceValue)
+            and not self.initial.has_value
+        ):
+            return None
+        else:
+            return self.initial
 
-        field_type = instance.default_field
+    def get_image_field(self):
+        self.kwargs["widget"] = FlexibleImageWidget(
+            help_text=self.help_text,
+            user=self.user,
+            current_value=self.initial,
+            # also passing the CIV as current value here so that we can
+            # show the image name to the user rather than its pk
+        )
+        upload_queryset = get_objects_for_user(
+            self.user,
+            "uploads.change_userupload",
+        ).filter(status=UserUpload.StatusChoices.COMPLETED)
+        image_queryset = get_objects_for_user(self.user, "cases.view_image")
+        return FlexibleImageField(
+            upload_queryset=upload_queryset,
+            image_queryset=image_queryset,
+            **self.kwargs,
+        )
 
-        if instance.is_image_kind:
-            kwargs["widget"] = FlexibleImageWidget(
-                help_text=help_text,
-                user=user,
-                current_value=initial,
-                # also passing the CIV as current value here so that we can
-                # show the image name to the user rather than its pk
-            )
-            upload_queryset = get_objects_for_user(
-                user,
-                "uploads.change_userupload",
-            ).filter(status=UserUpload.StatusChoices.COMPLETED)
-            image_queryset = get_objects_for_user(user, "cases.view_image")
-            self._field = field_type(
-                upload_queryset=upload_queryset,
-                image_queryset=image_queryset,
-                **kwargs,
+    def get_json_field(self):
+        field_type = self.instance.default_field
+        default_schema = {
+            **INTERFACE_VALUE_SCHEMA,
+            "anyOf": [{"$ref": f"#/definitions/{self.instance.kind}"}],
+        }
+        if field_type == forms.JSONField:
+            self.kwargs["widget"] = JSONEditorWidget(schema=default_schema)
+        self.kwargs["validators"] = [
+            JSONValidator(schema=default_schema),
+            JSONValidator(schema=self.instance.schema),
+        ]
+        extra_help = ""
+        return field_type(
+            help_text=_join_with_br(self.help_text, extra_help), **self.kwargs
+        )
+
+    def get_file_field(self):
+        key = f"value_type_{self.instance.slug}"
+        # on JobCreateForm interfaces are prepended with underscore
+        alt_key = f"value_type__{self.instance.slug}"
```

Comment:
We should be consistent, no?

Reply:
Yes, I agree that we should be consistent, but at the time when I added the underscore my thought process was the following: adding an underscore to all interface fields everywhere would be consistent but touches a lot more of the codebase than is strictly necessary for this PR, and so it seemed better to only add the underscore where it is really necessary (i.e. on JobCreation where we have other fields that could name clash with interface names - this is not (yet) an issue for archive items or display sets since they only have interface fields).

Reply:
Having said that, I do agree that only adding the underscore for job creation is weird and adds awkward work-around code snippets in a few places. Moreover, it does prevent future issues for archive items and display sets if we ever add fields to those forms where potential name clashes could happen. Maybe the way to go about this is to add the underscore everywhere in a separate PR.

Reply:
Why not use a proper prefix rather than just an underscore?

Reply:
What do you mean by proper prefix? How would that be different from just an underscore?

Reply:
I suggested `interface_form_field_prefix__` in another comment - then it is obvious to future developers what this is there for.

Reply:
Ok, I'll use that. But do we want to stay with the current implementation and only add this prefix on the JobCreateForm, or should I go ahead and use this everywhere (with all the necessary code changes in other places)?